### PR TITLE
feat: Glob utilities and misc functions

### PR DESCRIPTION
**Pull request by leongdl**
### What was the problem/requirement? (What/Why)
- We are introducing Job Attachment Manifest and Attachments APIs.
- There will be snapshot, diff, upload and download functionalities to create and manage manifests.
- There will be upload and download attachment of assets captured by manifests.
 
### What was the solution? (How)
- This PR is the first part of submitting
 - https://github.com/aws-deadline/deadline-cloud/pull/446
- In this PR, I am introducing utilities for Globbing file systems, processing a glob config file and misc utility functions.
 
### What is the impact of this change?
- Utilities to get Job Attachments user interfaces launched.

### How was this change tested?
- hatch build, hatch run fmt, hatch run lint, hatch run test.
- Integration tests are not applicable here, and will be added with subsequent CLI / API tests.

### Was this change documented?
- These are internal classes.

### Is this a breaking change?
No.

### Does this change impact security?
- No, these are internal utilities. Standard AWS security review best practices.

### CODE REVIEW

**Review by benl-2023 — APPROVED** [review]

File: src/deadline/job_attachments/_glob.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+from pathlib import Path
+from typing import List, Optional
+from deadline.client.exceptions import NonValidInputError
+from deadline.job_attachments.models import GlobConfig
+
+
+def _process_glob_inputs(glob_arg_input: str) -> GlobConfig:
+    """
+    Helper function to process glob inputs.
+    glob_input: String, can represent a json, filepath, or general include glob syntax.
```
Comment (mwiebe) [inline src/deadline/job_attachments/_glob.py:13]: Is the plan here to accept JSON as a CLI parameter input? If so, maybe we can adjust that to accept multiple arguments of include & exclude instead? I believe CLI interfaces that don't use JSON that way are generally more comfortable to use.

File: src/deadline/job_attachments/_glob.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        raise NonValidInputError(f"Glob input {glob_arg_input} cannot be deserialized as JSON")
+
+    return glob_config
+
+
+def _glob_paths(
+    path: str, include: List[str] = ["**/*"], exclude: Optional[List[str]] = None
+) -> List[str]:
+    """
+    Glob routine that supports Unix style pathname pattern expansion for includes and excludes.
```
Comment (mwiebe) [inline src/deadline/job_attachments/_glob.py:48]: This description doesn't say what the function does. What is the purpose of this function, what does it produce as output?

**Review by godobyte — APPROVED** [review]

File: src/deadline/job_attachments/_glob.py (hunk 1 above)
Comment (leongdl) [reply src/deadline/job_attachments/_glob.py:13]: Do you mean something like 
```--include abc/*.xyz,def/*.123```. ?

I was going by the OpenJD suggestion for globs, but happy to go about it either ways.

File: src/deadline/job_attachments/_glob.py (hunk 2 above)
Comment (leongdl) [reply src/deadline/job_attachments/_glob.py:48]: Will add more details!

File: src/deadline/job_attachments/_glob.py (hunk 1 above)
Comment (leongdl) [reply src/deadline/job_attachments/_glob.py:13]: I was thinking of something like this example: https://github.com/aws-deadline/deadline-cloud/pull/446/files#diff-33ab9cc270d578c6903c5e1cf7eb1e25e518cfc7222717f4599df29a368d0cc5R8

Comment (leongdl) [reply src/deadline/job_attachments/_glob.py:13]: For flexibility, I can also add `--include` and `--exclude` as comma separated inputs w/ the CLI version too. This function here specifically supports JSON file and JSON string.

Comment (leongdl) [reply src/deadline/job_attachments/_glob.py:13]: We chatted internally within the team. This code provides support for a JSON based glob syntax and can be used for large lists of include and exclude. For CLI use cases, I will also add `--include` and `--exclude` so both basis are covered.

File: src/deadline/job_attachments/_glob.py (hunk 2 above)
Comment (leongdl) [reply src/deadline/job_attachments/_glob.py:48]: Updated in the next PR of this feature series.

**Pull request merged**